src/main/java/org/jboss/staxmapper/XMLMapper.java

```diff
@@ -124,5 +124,20 @@ private Factory() {
         public static XMLMapper create() {
             return new XMLMapperImpl();
         }
+
+        /**
+         * Convenience method that Creates a new mapper instance pre-registered with a set of schemas, each supplying an element reader.
+         * @param <T> the operating type of the reader
+         * @param <S> the schema type
+         * @param schemas the schemas with which the returned mapper will be auto-registered
+         * @return a new mapper instance, pre-configured with the specified schemas
+         */
+        public static <T, S extends Schema & Supplier<XMLElementReader<T>>> XMLMapper create(Iterable<S> schemas) {
```

Comment:
Is a `Schema` always expected to also be a `Supplier<XMLElementReader<?>>`? This seems at odds with the above JavaDoc example, where a `Schema` was more expected to map to an `XMLElementReader` by way of a secondary function.

I would at least consider either changing the `XMLElementReader` to be more schema-aware, or make `Schema` more of a data object.

Reply:
@dmlloyd  I reworked this a bit based on your feedback.  Specifically:
* Split-up Schema into 2 interfaces:
  1. Namespace - constructs a namespace URI.  Added type parameter to narrow usage of Namespace.since(...) to classes of the same type.
  1. QNameProvider - creates a QName from a Namespace and a local name
* Added a separate XMLElementReaderFactory interface that supplies a reader for a specific element QName
* Replaced previous XMLMapper.Factory.create(...) with a XMLMapper.registerRootElement(...) convenience method